### ISSUE TITLE
Add test folder to package source distribution

### ISSUE THREAD

**Issue by j-emberton** [ImperialCollegeLondon/pyrealm#259]
**Is your feature request related to a problem? Please describe.**

Pyrealm is distributed from PyPI without the tests included. This means that anyone who wants to build Pyrealm themselves doesn't have direct access to the tests to verify their build.

This is relevant to people and orgs (e.g. anaconda etc) who like to build their own packages.

**Describe the solution you'd like**

Add the tests to the source distribution

**Comment by davidorme** [comment]
That seems sensible - I hadn't really looked at what is included in the source distribution. I explicitly included the `pyrealm_build_data` package in source, which is excluded from the binary distributions, but it doesn't include anything else needed to build and test the package, so the source distribution doesn't really stand alone.

It's not just tests, really - the docs should probably go in too?

**Comment by j-emberton** [comment]
Yes I'd agree documentation would be a useful addition.

**Comment by j-emberton** [comment]
@davidorme @MarionBWeinzierl, It was mentioned by one of our reviewers yesterday that they had tried to follow the worked examples in the Pyrealm notes but that the distribution lacks the `build data` that some of the worked examples use. They managed to figure it out by going direct to GitHub, but we should probably rectify this for v2.0

With your blessing I'll add this issue to the v2.0 milestone

**Comment by davidorme** [comment]
There's a difference between including more of the code in the `source` distribution (`tests` and `docs` probably) and making it easier for _users_ to follow along with the worked examples. I don't know how we do that - could they install the `pyrealm_build_data` package separately to be able to use that?

**Comment by davidorme** [comment]
I've looked around a bit and I can't see an easy route for the package structure at the moment. I looked at the package `extras` functionality, thinking that we could do something like `pip install pyrealm[data]` but I think that only applies to extra package dependencies. You can include package data, but I don't think that is optional and I'd rather not bloat the standard binary install.

Perhaps we need to split `pyrealm_build_data` out into a standalone data package, but that complicates the docs etc. Or I've missed something else 😄

**Comment by MarionBWeinzierl** [comment]
Could we add a sentence or two to the README/docs that says that for running the tests they have to download the data from the Github repository? That would not change anything about the package but give users the info to run the tests without having to do any digging around in the repository.

**Comment by davidorme** [comment]
Can we create data releases in parallel to package releases? It would be good to be able to point to a zip file on Github or Zenodo. Maybe have some kind of manually triggered CI workflow that bundles the data somewhere?

**Comment by j-emberton** [comment]
> Can we create data releases in parallel to package releases? It would be good to be able to point to a zip file on Github or Zenodo. Maybe have some kind of manually triggered CI workflow that bundles the data somewhere?

I think the simplest thing is to include the bare minimum data for the worked examples within the docs folder, and leave everything else in build data. Anything needed for the worked examples would then be included in the binaries. We can always trim these down to sensible size (fewer numbers and lower precision?) to they don't burden the package.

How does this sound?

A robust solution for the build data would be zenodo. We'd have to then create fixtures to download the data for each test session which would slow down local test turnaround.

**Comment by davidorme** [comment]
I'm not sure that solves the issue. The binary package does not include the `docs` folder so users will be no more able to play along with the examples than at the moment. We could include some small data files as package data in the binary but then we constrain our examples in docs to smaller, less illustrative inputs and add size to the package. 

I think the current setup of using the build data for regression testing and more illustrative testing works well, but the problem is really just distributing that data for interested users. As @MarionBWeinzierl says, we can just provide links to clone the files, but we could also maintain a separate Zenodo package of the data to make life easy. I don't think that should _replace_ the `pyrealm_build_data` setup, just provide an easy download for users.

**Comment by j-emberton** [comment]
OK, that seems reasonable.

**Comment by davidorme** [comment]
One thing we could easily (?) do is to add a step to our release CI that zips up `pyrealm_build_data` and adds it as an artifact to the release files. That way we get an updated single downloadable file associated with each release?

**Comment by davidorme** [comment]
Of course, that's already included in the source zip and tar.gz, but it would be more user friendly. Zenodo would be better though.

**Comment by j-emberton** [comment]
OK, I'll reformat this issue as a packaging/distro/data meta issue and break out these tings as separate tasks
